packages/utilities/src/lib/sleep.ts

```diff
@@ -0,0 +1,77 @@
+export interface SleepOptions {
+	/**
+	 * When provided the corresponding `AbortController` can be used to cancel an asynchronous action.
+	 */
+	signal?: AbortSignal | undefined;
+
+	/**
+	 * Set to `false` to indicate that the scheduled `Timeout`
+	 * should not require the Node.js event loop to remain active.
+	 * @default true
+	 */
+	ref?: boolean | undefined;
+}
+
+export class AbortError extends Error {
+	public code: string;
+	public constructor(
+		message?: string,
+		options?: {
+			cause?: unknown;
+			code?: string;
+		}
+	) {
+		super(message, options);
+		this.name = 'AbortError';
+		this.code = options?.code ?? 'ERR_ABORT';
+	}
+}
```

Comment:
Where does this come from?

Reply:
`timers/promise` trows `AbortError` with cause `DOMException`. So I was trying to make it similar error.

Reply:
Interestingly enough, `timers/promise` sets `cause` as a string, not as a `DOMException`. See below:

https://github.com/nodejs/node/blob/2a4452a53af65a13db4efae474162a7dcfd38dd5/lib/timers/promises.js#L83-L87
https://github.com/nodejs/node/blob/2a4452a53af65a13db4efae474162a7dcfd38dd5/lib/timers/promises.js#L44-L49
https://github.com/nodejs/node/blob/2a4452a53af65a13db4efae474162a7dcfd38dd5/lib/internal/errors.js#L843-L852

And I sorta agree, an exception inside an exception is inefficient and unreadable.

Reply:
Yeah after your previous comment, I added `signal.reason` and found that if we don't add a reason, it's automatically `DOMException`.  So changed it to match `timers/promisify`